### PR TITLE
Typo corrected and new events added

### PR DESCRIPTION
**Pull request by jordiadell**
I need this to be merged into the main branch. This will imply changes in robot, so I need to be informed when the merge is done.

I corrected a typo in listen_state variable
and added the following events:
FAILED_DECODING
CALIBRATION

### CODE REVIEW

**Comment by awesomebytes** [comment]
As a note, the ENUMs have changed name-to-value. Shouldn't affect anyone if it's really using the ENUMs but I'm just warning.

**Comment by jordiadell** [comment]
You should NEVER use the integer value of an enum. I check all the code that works with these enums and we are not going to have any problem with this. As they are referenced always by the name rather than their value. So we can go on with the merge

File: pal_interaction_msgs/msg/ASREvent.msg [hunk 1]
```diff
@@ -15,9 +16,10 @@ int8 EVENT_ASR_ACTIVATION     = 3
...
 ## Listen states
 int8 LISTEN_UNDEFINED = 20
-int8 LISTEN_SPEECH_DETECTED = 21
+int8 LISTEN_CALIBRATION = 21
 int8 LISTEN_WAITING_FOR_SPEECH = 22
-int8 LISTEN_END_OF_SPEECH = 23
+int8 LISTEN_SPEECH_DETECTED = 23
+int8 LISTEN_END_OF_SPEECH = 24
 
 # 21-> Some speech has been detected.
```
Comment (po1) [inline pal_interaction_msgs/msg/ASREvent.msg]: this comment doesn't seem right anymore, shouldn't it be fixed?

**Comment by po1** [comment]
Also, it **will** affect people in the following cases:
- data stored in bagfiles
- binaries (libraries and executable) will need to be rebuilt

This is similar to an ABI break, which means that everything that depends on this message should be rebuilt, but that the code should not require any modifications.

**Comment by jordiadell** [comment]
I really do no think anyone store a bag file with /asr_event topic. 
I know that this commit breaks API, AND ABI. But this is the time to do it, since this ROS API was recently born.

**Comment by po1** [comment]
It does not break API. But don't worry, you are right. Very few people rely on this outside of us (is there a single person?)

EDIT: it does break it (hadn't see the typo for `listen_state`)

**Comment by po1** [comment]
@jordiadell I am going to merge this. Are you ready to commit the fix in robot?

**Comment by jordiadell** [comment]
Yes I do.

**Comment by po1** [comment]
So interactive.

**Pull request merged**